packages/wallet-sdk/src/sign/scw/utils/createSubAccountSigner.ts

```diff
@@ -52,74 +68,204 @@ export async function createSubAccountSigner({ chainId }: { chainId: number }) {
+          method: 'wallet_prepareCalls',
+          params: [
+            {
+              calls: args.params[0].calls as {
+                to: Address;
+                data: Hex;
+                value: Hex;
+              }[],
+              chainId: chainId,
+              from: subAccount.address,
+              capabilities:
+                'capabilities' in args.params[0]
+                  ? (args.params[0].capabilities as Record<string, any>)
+                  : {},
+            },
+          ],
+        })) as PrepareCallsSchema['ReturnType'];
+
+        const signature = await owner!.sign?.({
+          // Hash returned from wallet_prepareCalls is double hex encoded
+          hash: hexToString(prepareCallsResponse.signatureRequest.hash) as `0x${string}`,
+        });
+
+        let signatureData: SendPreparedCallsSchema['Parameters'][0]['signature'];
+
+        if (!signature) {
+          throw standardErrors.rpc.internal('signature not found');
         }
-        case 'eth_signTypedData_v4': {
-          assertArrayPresence(args.params);
-          return account.signTypedData(args.params[1] as TypedDataDefinition);
+
+        if (isHex(signature)) {
+          signatureData = {
+            type: 'secp256k1',
+            data: {
+              address: owner!.address!,
+              signature: signature as Hex,
+            },
+          };
+        } else {
+          const { webauthn, signature: signatureHex } = signature;
+
+          const signatureRaw = Signature.fromHex(signatureHex);
+
+          signatureData = {
+            type: 'webauthn',
+            data: {
+              signature: JSON.stringify(
+                convertCredentialToJSON({
+                  id: owner!.id ?? '1',
+                  rawId: owner!.id ?? '',
+                  response: {
+                    authenticatorData: hexToBytes(webauthn.authenticatorData),
+                    clientDataJSON: webauthn.clientDataJSON,
+                    signature: asn1EncodeSignature(signatureRaw.r, signatureRaw.s),
+                  },
+                  type: JSON.parse(webauthn.clientDataJSON).type,
+                })
+              ),
+              publicKey: owner!.publicKey,
+            },
+          };
         }
-        case 'eth_signTypedData_v1':
-        case 'eth_signTypedData_v3':
-        case 'wallet_addEthereumChain':
-        case 'wallet_switchEthereumChain':
-        default:
-          throw standardErrors.rpc.methodNotSupported();
+
+        const sendPreparedCallsResponse = (await request({
+          method: 'wallet_sendPreparedCalls',
+          params: [
+            {
+              version: '1.0',
+              type: prepareCallsResponse.type,
+              data: prepareCallsResponse.userOp,
+              chainId: prepareCallsResponse.chainId,
+              signature: signatureData,
+            },
+          ],
+        })) as SendPreparedCallsSchema['ReturnType'];
+
+        return sendPreparedCallsResponse[0];
```

Comment:
This response doesn't map to anything. I think it might be the signature vs the user op hash

Reply:
weird the userop hash for my test was `0x1c91c7eae34080db52cdcb7abdaf1e5f05d27247d0ce99444692c203b19bcd57` but the response returned `0x1c91c7eae34080db52cdcb7abdaf1e5f05d27247d0ce99444692c203b19bcd570000000000000000000000000000000000000000000000000000000000014a34`

Reply:
afaik that is an arbitrary id that needs to be awaited using wallet_getCallsStatus which will return receipts

Reply:
Yeah it should be returning the correct callId. I'd like to index on this a bit more to make sure that we're always returning the properly formatted call id. Lets talk about it in stand up